core/chains/evm/utils/utils.go

```diff
@@ -35,3 +181,108 @@ func NewRedialBackoff() backoff.Backoff {
 	}
 
 }
+
+// RetryWithBackoff retries the sleeper and backs off if not Done
+func RetryWithBackoff(ctx context.Context, fn func() (retry bool)) {
```

Comment:
This might be useful for common and other chains

Reply:
`RetryWithBackoff` uses a Backoff with arbitrary values ( Min: 1 * time.Second, Max: 10 * time.Second ) so it's better if we keep it implementation-specific